### PR TITLE
Migrate gadgets from zkevm-circuits mod to it'w own crate

### LINKED ISSUE #444: Create a crate for commonly used gadgets
### Problem

- [zkevm-circuits/src/gadget](https://github.com/appliedzkp/zkevm-circuits/tree/main/zkevm-circuits/src/gadget) currently contains 3 gadgets: evm_word/monotone/is_zero. evm_word and monotone are unused anywhere, but is_zero is used in bytecode unroller. 
- In the keccak we need the is_zero for the padding validation.

### Proposed Solution

1. Move the gadget as a separate crate, but still in zkevm-circuits repo
2. The gadget crate can maintain some commonly used circuits. is_zero is definitely useful everywhere. Another candidate would be the more general purposed running sum.
3. Remove evm_word and monotone if we don't use them in the future.

### PR DESCRIPTION
**Pull request by CPerezz**
Resolves: #444

### CODE REVIEW

**Review by roynalnaruto — APPROVED** [review]
Looks good to me!

**Review by miha-stopar — APPROVED** [review]
Seems a good idea to me! It seems all ok to me!

File: gadgets/src/is_zero.rs [hunk 1]
```diff
@@ -18,19 +26,41 @@ pub(crate) trait IsZeroInstruction<F: FieldExt> {
...
+    /// +----+-------+-----------+-----------------------+---------------------------------+-------------------------------------+
+    /// | V  | 0     | 0         | 1                     | 0
+    /// | 0                                   | |    | 0     | x         | 1
+    /// | 0                               | x                                   |
+    /// |    | x     | 0         | 1                     | x
+    /// | 0                                   | | V  | x     | 1/x       | 0
+    /// | 0                               | 0                                   |
+    /// |    | x     | y         | 1 - xy                | x(1 - xy)
+    /// | y(1 - xy)                           |
+    /// +----+-------+-----------+-----------------------+---------------------------------+-------------------------------------+
```
Comment (ChihChengLiang) [inline gadgets/src/is_zero.rs:63]: @CPerezz the formatting seems broken here

Comment (CPerezz) [reply gadgets/src/is_zero.rs:63]: Ohhh damn!!! 
Let me make a quick fix PR

**Pull request merged**